lib/Common/Memory/CollectionState.h

```diff
@@ -91,3 +97,43 @@ enum CollectionState
     CollectionStatePostSweepRedeferralCallback = Collection_PostSweepRedeferralCallback,
     CollectionStatePostCollectionCallback = Collection_PostCollectionCallback,
 };
+
+class CollectionStateChangedObserver
+{
+public:
+    virtual void CollectionStateChanged(CollectionState newVal, CollectionState oldVal) = 0;
+};
+
+class ObservableCollectionState
```

Comment:
I like this but I think you can make it generic and move it to Common/Common since I can imagine other places in the engine where having an ObservableField would be useful.